internal/services/managedredis/managed_redis_resource.go

```diff
@@ -0,0 +1,766 @@
+									Type:     pluginsdk.TypeString,
+									Optional: true,
+								},
+
+								"version": {
+									Type:     pluginsdk.TypeString,
+									Computed: true,
+								},
+							},
+						},
+					},
+
+					"port": {
+						Type:         pluginsdk.TypeInt,
+						Optional:     true,
+						Default:      10000,
+						ValidateFunc: validation.IntBetween(0, 65353),
+					},
+
+					"primary_access_key": {
+						Type:      pluginsdk.TypeString,
+						Computed:  true,
+						Sensitive: true,
+					},
+
+					"secondary_access_key": {
+						Type:      pluginsdk.TypeString,
+						Computed:  true,
+						Sensitive: true,
+					},
+				},
+			},
+		},
+
+		"high_availability_enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+			ForceNew: true,
+			Default:  true,
+		},
+
+		"identity": commonschema.SystemAssignedUserAssignedIdentityOptional(),
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r ManagedRedisResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"hostname": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (r ManagedRedisResource) ModelObject() interface{} {
+	return &ManagedRedisResourceModel{}
+}
+
+func (r ManagedRedisResource) ResourceType() string {
+	return "azurerm_managed_redis"
+}
+
+func (r ManagedRedisResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return redisenterprise.ValidateRedisEnterpriseID
+}
+
+func (r ManagedRedisResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			clusterClient := metadata.Client.ManagedRedis.Client
+			dbClient := metadata.Client.ManagedRedis.DatabaseClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var model ManagedRedisResourceModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			// Check existing cluster / db
+
+			clusterId := redisenterprise.NewRedisEnterpriseID(subscriptionId, model.ResourceGroupName, model.Name)
+
+			existingCluster, err := clusterClient.Get(ctx, clusterId)
+			if err != nil {
+				if !response.WasNotFound(existingCluster.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", clusterId, err)
+				}
+			}
+
+			if !response.WasNotFound(existingCluster.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), clusterId)
+			}
+
+			dbId := databases.NewDatabaseID(subscriptionId, clusterId.ResourceGroupName, clusterId.RedisEnterpriseName, DefaultDatabaseName)
+
+			existingDb, err := dbClient.Get(ctx, dbId)
+			if err != nil {
+				if !response.WasNotFound(existingDb.HttpResponse) {
+					return fmt.Errorf("checking for presence of existing %s: %+v", dbId, err)
+				}
+			}
+
+			if !response.WasNotFound(existingDb.HttpResponse) {
+				return fmt.Errorf("expected db %s to be not found: %+v", dbId, err)
+			}
```

Comment:
This suggests that we're concerned that the Service will create the Default DB on Cluster creation?  We should get confirmation from the Service Team that this is not the case, or we could end up with an unusable resource here? 

Also, since we're using a `PUT` on the DB Create method, we can just go ahead and over-write the DB anyway since we just created a new cluster so we want to continue regardless?

```suggestion
```